resources/assets/lib/beatmap-discussions/editor.tsx

```diff
@@ -247,21 +252,23 @@ export default class Editor extends React.Component<Props, State> {
   }
 
   post = () => {
-    this.setState({posting: true}, () => {
-      this.xhr = $.ajax(laroute.route('beatmapsets.discussion.review', {beatmapset: this.props.beatmapset.id}), {
-        data: { document: this.serialize() },
-        method: 'POST',
+    if (this.showConfirmationIfRequired()) {
+      this.setState({posting: true}, () => {
+        this.xhr = $.ajax(route('beatmapsets.discussion.review', {beatmapset: this.props.beatmapset.id}), {
+          data: {document: this.serialize()},
+          method: 'POST',
+        });
+
+        this.xhr.then((data) => {
```

Comment:
The indentation seems a bit weird on `.always` and `.catch`.

Additionally, `.done` instead of `.then` and `.fail` instead of `.catch` should return `JQueryXHR` so everything can be chained during initial assignment `this.xhr = `.